### PR TITLE
Simplify debug machinery and print line/function

### PR DESCRIPTION
**Pull request by simo5**
Fixes: #149

### CODE REVIEW

File: src/debug.h [hunk 1]
```diff
@@ -5,38 +5,47 @@
...
+        } \
+    } while (0)
 
 #define P11PROV_debug_slot(...) \
-    P11PROV_debug_status(p11prov_debug_slot(__VA_ARGS__))
+    do { \
+        if (debug_level < 0) { \
+            p11prov_debug_init(); \
+        } else if (debug_level > 0) { \
+            p11prov_debug_slot(__VA_ARGS__); \
```
Comment (Jakuje) [inline src/debug.h:33]: Should not this be separate `if` (similarly in the other macros here)? Isn't this going to ignore the first message and do just the initialization?

Comment (simo5) [reply src/debug.h:33]: good catch!

Comment (simo5) [reply src/debug.h]: Fixed in rebased patch

**Review by Jakuje — APPROVED** [review]
lgtm

**Pull request merged**